Automattic/jetpack
# Fix sync tests on latest WP Develop and VVV

Fixes many issues with `--testsuite=sync` on VVV with latest wordpress-develop branch.

#### Changes proposed in this Pull Request:

* Set `WP_PLUGIN_DIR` and `ABSPATH` in boostrap.php so that plugin installation and update tests pass. Previously they were being set to the values for the `wordpress-develop` install (which hosts the test classes) but I wanted them to point to the `wordpress-default` install, where Jetpack is being developed.
* Replace reference to require `$test_root . '/includes/listener-loader.php';` with `speed-trap-listener.php`. This file appears to have been renamed recently.
* Don't run `test_add_whitelisted_option_on_init_89` if `init` has already run - it's almost guaranteed to fail. Warn instead so we can fix that test properly in the future.
* Add new `"jetpack_filter_${transient}_value"` filter which lets us inject a fake update into the list of available WP versions, just in case there are no updates against the WP version you're testing on.

#### Testing instructions:
* run `phpunit --testsuite=sync` on a VVV install, with this branch checked out in the `wordpress-default` branch. `wordpress-default` and `wordpress-develop` installs should both be as up-to-date as possible.

#### Proposed changelog entry for your changes:

* Improve test reliability on VVV
